lib/src/http_request_impl.dart

```diff
@@ -7,7 +7,7 @@ class HttpRequestImpl implements HttpRequest {
       : dio = Dio(BaseOptions(
           baseUrl: serverUrl,
           headers: <String, dynamic>{
-            if (apiKey != null) 'X-Meili-API-Key': apiKey,
+            if (apiKey != null) 'Authorization': 'Bearer ${apiKey}',
```

Comment:
I totally forgot to say during this PR that it's always better to cut the PR with the big breaking changes one by one. In the idea to be able for the contributors and for the developers to follow correctly the changes and to make roll-back if needed, even if it happens very rarely. For example, this change should have been single.
You could not know but we had a meeting about this changing but you can find the information [here](https://github.com/meilisearch/integration-guides/issues/157#issuecomment-992630594)

Reply:
Thanks for the tip! Next time I would reduce the quantity of the changes or just make it more clear :D

Either way, I have created multiple PRs like #121 (who dealt with this breaking change).